spec/support/json_controller_helper.rb

```diff
@@ -1,91 +1,93 @@
 shared_examples 'a JSON resource controller:' do
 
-  # singular downcase string of the class name
-  let(:model_string) { model_class.to_s.underscore }
+  # singular downcase string of the factory name
+  let(:factory_name) { factory.to_s }
 
-  # singular downcase symbol of the class name
-  let(:model_symbol) { model_string.to_sym }
+  # minimal set of attributes required to create a valid model
+  let(:valid_attributes) { attributes_for factory }
 
   # minimal set of attributes required to create an invalid model
   let(:invalid_attributes) do
-    attributes_for "invalid_#{model_string}".to_sym
+    attributes_for "invalid_#{factory_name}".to_sym
   end
 
   # set of valid attributes required to update model
   let(:updated_valid_attributes) do
-    attributes_for "updated_valid_#{model_string}".to_sym
+    attributes_for "updated_valid_#{factory_name}".to_sym
   end
 
   # set of invalid attributes required to update model
   let(:updated_invalid_attributes) do
-    attributes_for "invalid_#{model_string}".to_sym
+    attributes_for "invalid_#{factory_name}".to_sym
   end
 
   # symbol of the variable being assigned by the
   # controller in all the other methods
-  let(:assigned_variable) { model_symbol }
+  let(:assigned_variable) { factory }
 
   # symbol of the variable being assigned by the RESTful
   # controller in the index method
-  let(:mass_assigned_variable) { model_string.pluralize.to_sym }
+  let(:mass_assigned_variable) { factory_name.pluralize.to_sym }
+
+  let(:model) { FactoryGirl.create(factory) }
 
   describe 'GET index' do
     it 'assigns all models as @models' do
-      model = model_class.create! valid_attributes
-      get :index, request_params, valid_session
+      get :index, request_params(model), valid_session
       assigns(mass_assigned_variable).should eq([model])
     end
   end
 
   describe 'GET show' do
     it 'assigns the requested model as @model' do
-      model = model_class.create! valid_attributes
-      get :show, request_params(id: model.to_param), valid_session
+      get :show, request_params(model, id: model.to_param), valid_session
       assigns(assigned_variable).should eq(model)
     end
   end
 
   describe 'POST create' do
     describe 'with valid params' do
-      it 'creates a new model_class' do
+      it 'creates a new model' do
+        sample_model = FactoryGirl.build(factory)
         expect do
           post :create,
-               request_params(assigned_variable => valid_attributes),
+               request_params(nil, assigned_variable => valid_attributes),
                valid_session
-        end.to change(model_class, :count).by(1)
+        end.to change(sample_model.class, :count).by(1)
       end
 
-      it 'assigns a newly created course as @course' do
+      it 'assigns a newly created model as @model' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.